### PR TITLE
Add many basic instructions. This also required us refine many of the definitions in common.lean.

### PR DESCRIPTION
**Pull request by dagit**
Please let me know if you'd like changes to anything. In the future I'll try to work in smaller chunks, but in this case it was nice to make these changes together to reduce overall churn.

### CODE REVIEW

**Comment by dagit** [comment]
Ping!

Is there anything I can do to help move along the review on this?

Thanks.

File: lean/common.lean [hunk 1]
```diff
@@ -57,26 +59,32 @@ protected def do_add : nat_expr → nat_expr → nat_expr
 | (lit x) (lit y) := lit (x+y)
 | x y := add x y
 
-protected def do_sub : nat_expr → nat_expr → nat_expr
```
Comment (joehendrix) [inline lean/common.lean]: Did you mean to delete "protected"?

File: lean/common.lean [hunk 2]
```diff
@@ -276,6 +284,8 @@ def spl := reg8l 4
 def bpl := reg8l 5
 def sil := reg8l 6
 def dil := reg8l 7
+ -- TODO: this order okay?
+def ah  := reg8l 8
```
Comment (joehendrix) [inline lean/common.lean]: flexdis uses the convention that the high registers are 16 + the low register index.  So ah should be 16. 

I'd write ` def reg8h (i:fin 16) := lhs.reg $ reg.concrete_gpreg (16+i) gpreg_type.reg8l`, then `def ah := reg8h 0`.

File: lean/common.lean [hunk 3]
```diff
@@ -297,6 +307,18 @@ def rax := reg64 0
 def rcx := reg64 1
 def rdx := reg64 2
 def rbx := reg64 3
+def rsi := reg64 4
+def rdi := reg64 5
+def rsp := reg64 6
+def rbp := reg64 7
```
Comment (joehendrix) [inline lean/common.lean]: I prefer the numbering scheme used in encoding registers in machine code, so it should be:

```
def rsp := reg64 4
def rbp := reg64 5
def rsi := reg64 6
def rdi := reg64 7
```

See http://www.c-jump.com/CIS77/CPU/x86/X77_0060_mod_reg_r_m_byte.htm

**Comment by dagit** [comment]
I fixed those things and one more typo I had missed previously.

Please let me know if there are other changes you would like.

**Pull request merged**